File: BestTimeToBuyAndSellStock.java
import java.util.*;

class BestTimeToBuyAndSellStock {
    public int maxProfit(int[] prices) {
        int min = Integer.MAX_VALUE, max = 0;
        for (int i=0; i<prices.length; i++) {
            int price = prices[i];
            if (price < min) min = price;
            int profit = price - min;
            if (profit > max) max = profit;
        }
        return max;
    }
}

File: neetcode/BestTimeToBuyAndSellStock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minP=prices[0],maxP=0;
        for(int i=1; i<prices.size(); i++) {
            maxP=max(maxP,prices[i]-minP);
            minP=min(minP,prices[i]);
        }
        return maxP;
    }
};
